hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/RequestFactoryImpl.java

```diff
@@ -702,6 +719,13 @@ public static final class RequestFactoryBuilder {
      */
     private boolean isMultipartUploadEnabled = true;
 
+    /**
+     * Timeout for uploading objects/parts.
+     * This will be set on data put/post operations only.
+     * A zero value means "no custom timeout"
+     */
+    private Duration partUploadTimeout = Duration.ZERO;
```

Comment:
I believe you have tested this explicitly. Hopefully zero doesn't mean infinite.

Reply:
I'll set to the default, even though I've checked that wherever the builder is currently used it now fills in the field

Reply:
yes, I will set to the default...even though its set up properly in production, there are some test cases which didn't